x/incentive/keeper/genesis.go

```diff
@@ -49,6 +51,80 @@ func (k Keeper) InitGenesis(ctx context.Context, gs types.GenesisState) error {
 		}
 	}
 
+	for _, entry := range gs.RefundableMsgHashes {
+		// hashes are hex encoded for better readability
+		bz, err := hex.DecodeString(entry)
+		if err != nil {
+			return fmt.Errorf("error decoding msg hash: %w", err)
+		}
+		if err := k.RefundableMsgKeySet.Set(ctx, bz); err != nil {
+			return fmt.Errorf("error storing msg hash: %w", err)
+		}
+	}
+
+	for _, entry := range gs.FinalityProvidersCurrentRewards {
+		// check that fp address exists
+		fpAddr := sdk.MustAccAddressFromBech32(entry.Address)
+		acc := k.accountKeeper.GetAccount(ctx, fpAddr)
+		if acc == nil {
+			return fmt.Errorf("finality provider account with address %s does not exist", entry.Address)
+		}
+		if err := k.FinalityProviderCurrentRewards.Set(ctx, fpAddr.Bytes(), *entry.Rewards); err != nil {
```

Comment:
lets avoid going to the collections property `k.FinalityProviderCurrentRewards` and centralize the access to the collections by using `setFinalityProviderCurrentRewards`

In fact, these collections almost should be private, what do you think?

Reply:
done